Cargo.toml

```
@@ -47,17 +47,23 @@ name = "_pydantic_core"
 crate-type = ["cdylib", "rlib"]
 
 [features]
-# must be disabled for cargo bench
-extension-module = ["pyo3/extension-module"]
 # required for cargo bench
 auto-initialize = ["pyo3/auto-initialize"]
-default = ["mimalloc", "extension-module", "mimalloc/local_dynamic_tls", "pyo3/generate-import-lib"]
+default = ["mimalloc", "mimalloc/local_dynamic_tls", "pyo3/generate-import-lib"]
 
 [profile.release]
 lto = "fat"
 codegen-units = 1
 strip = true
 
+[dev-dependencies]
+# From https://doc.rust-lang.org/cargo/reference/specifying-dependencies.htm:
+# > Dev-dependencies are not used when compiling a package for building,
+# > but are used for compiling tests, examples, and benchmarks.
+# The following makes it so you don't need to add the --no-default-features --features auto-initialize
+# flags to `cargo test` or `cargo bench`.
+pydantic-core = { path = ".", default-features = false, features = ["auto-initialize"] }
```

Perhaps makes sense to delete the comment above this line. My novice self felt it made sense to provide the context but maybe this is obvious to more serious rust devs.

Based on this comment: https://github.com/PyO3/maturin/pull/1479#issuecomment-1428661399 I'm thinking it might make sense to replace this with:
```suggestion
pyo3 = { version = "0.18.2", features = ["auto-initialize"] }
```
but then the pyo3 version is listed twice. I'm okay either way, this is kind of getting outside the scope of what I know well.

---

Actually, it looks like this change actually be necessary for maturin to build properly since it doesn't recognize the path `"."`